test/compress/harmony.js

```diff
@@ -444,3 +476,66 @@ generators_yield_assign: {
     }
     expect_exact: "function*fn(){var x={};x.prop=yield 5}"
 }
+
+generator_yield_undefined: {
+    input: {
+        function* fn() {
+            yield;
+        }
```

Comment:
Undefined gets rewritten as `void 0`. Need to dive in the code generator and see if this statement could even be removed by default.